Intermittent crash in macOS CI, after test_secondary_window_with_content

This has now happened in 2 out of the last 3 merges to main:

* https://github.com/beeware/toga/actions/runs/9440671114/job/26000510273
* https://github.com/beeware/toga/actions/runs/9440341374/job/25999536477

And also a couple of weeks ago:

* https://github.com/beeware/toga/actions/runs/9296068601/job/25584229302

It always happens at the same point:

```
tests/test_window.py::test_title PASSED                                  [  9%]
tests/test_window.py::test_secondary_window[second_window_kwargs0] PASSED [  9%]
tests/test_window.py::test_secondary_window_with_args[second_window_kwargs0] PASSED [ 10%]
tests/test_window.py::test_secondary_window_with_content PASSED          [ 10%]

Test suite didn't report a result.
Saving log... started

Log saved to /Users/runner/work/toga/toga/testbed/logs/briefcase.2024_05_30-02_51_53.run.log

Saving log... done
Error: Process completed with exit code 200.
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -275,7 +275,7 @@ jobs:
     # Github runners seem to have intermittent connectivity issues.
     # See https://github.com/beeware/toga/issues/2632
     - name: Tune GitHub-hosted runner network
-      uses: smorimoto/tune-github-hosted-runner-network@v1
+      uses: smorimoto/tune-github-hosted-runner-network@v1.0.0
 
     - name: Checkout
       uses: actions/checkout@v4.1.6

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -272,6 +272,11 @@ jobs:
             sudo udevadm control --reload-rules
             sudo udevadm trigger --name-match=kvm
     steps:
+    # Github runners seem to have intermittent connectivity issues.
+    # See https://github.com/beeware/toga/issues/2632
+    - name: Tune GitHub-hosted runner network
+      uses: smorimoto/tune-github-hosted-runner-network@v1
+
     - name: Checkout
       uses: actions/checkout@v4.1.6
       with:

--- a/changes/2631.misc.rst
+++ b/changes/2631.misc.rst
@@ -1,0 +1,1 @@
+An intermittent failure in macOS CI was corrected.

--- a/changes/2632.misc.rst
+++ b/changes/2632.misc.rst
@@ -1,0 +1,1 @@
+An step has been added to the Github actions configuration to fix an issue with Github's network configuration.

--- a/testbed/tests/test_window.py
+++ b/testbed/tests/test_window.py
@@ -243,6 +243,9 @@ else:
             assert window_with_content.content == content
         finally:
             window_with_content.close()
+            await window_with_content_probe.redraw("Secondary window has been closed")
+            del window_with_content
+            gc.collect()
 
     async def test_secondary_window_cleanup(app_probe):
         """Memory for windows is cleaned up when windows are deleted."""

